File: read_pdf.js
const fs = require('fs');
const pdf = require('pdf-parse');

async function extractText(filePath) {
    let dataBuffer = fs.readFileSync(filePath);
    try {
        let data = await pdf(dataBuffer);
        return data.text;
    } catch (e) {
        return "Error reading " + filePath + ": " + e;
    }
}

async function main() {
    const mlResume = await extractText('./public/P_D_Dinuk_ML_Resume.pdf');
    const seResume = await extractText('./public/P_D_Dinuk_SE_Resume.pdf');
    fs.writeFileSync('resume_data.txt', "--- ML RESUME ---\n" + mlResume + "\n\n--- SE RESUME ---\n" + seResume);
    console.log("Extracted text saved to resume_data.txt");
}

main();

File: read_pdf.py
import PyPDF2

def extract_text(pdf_path):
    with open(pdf_path, 'rb') as f:
        reader = PyPDF2.PdfReader(f)
        text = ""
        for page in reader.pages:
            text += page.extract_text() + "\n"
        return text

with open('resume_data.txt', 'w', encoding='utf-8') as f:
    try:
        f.write("--- ML RESUME ---\n")
        f.write(extract_text('public/P_D_Dinuk_ML_Resume.pdf'))
    except Exception as e:
        f.write(str(e))
    try:
        f.write("\n\n--- SE RESUME ---\n")
        f.write(extract_text('public/P_D_Dinuk_SE_Resume.pdf'))
    except Exception as e:
        f.write(str(e))
